packages/dataviews/src/filters.js

```diff
@@ -85,16 +94,25 @@ const Filters = memo( function Filters( { fields, view, onChangeView } ) {
 		} ),
 	];
 
-	if ( filterComponents.length > 1 && view.type !== LAYOUT_LIST ) {
-		filterComponents.push(
-			<ResetFilters
-				key="reset-filters"
-				view={ view }
-				onChangeView={ onChangeView }
-			/>
-		);
+	// Reset should be hidden when either:
+	//
+	// - the layout is list
+	// - or there is only one primary filter and none secondary filters
+	if (
+		view.type === LAYOUT_LIST ||
+		( filters.length === 1 && primaryFilters === 1 )
```

Comment:
I think the logic here needs some small adjustment to take into account the total primary filters with the total filters in general. We could have more than one primary filter, etc..

Reply:
> only one primary

This rule feels a little bit magic. Basically, it means that this will work only for Patterns but implicitly so (Patterns only have one enum field and it's a primary one, while Pages have two enums but only one primary).

I would expect one of the following:

* Either the condition is consistent regardless of number (`there are only primary filters && all their values are undefined`), so that if I had a hypothetical version of Pages where Author were also a primary filter then the Reset button would disappear too.
* Or the rule to hide the Reset button is controlled elsewhere (by the consumer?).

I'm OK with giving it a go, I appreciate the experiment. I'm just pointing out what I think is an inconsistency or a special-casing. 

_edit:_ GitHub didn't refresh properly, but I see Nik has commented something similar.

Reply:
This is all the cases and how it works now:

1. There is at least a secondary filter: show reset.
2. List layout: do not show reset.
3. There are only primary filters, but there are many: show reset.
4. There is only one primary filter: do not show reset.

What would you want to do differently?

I prefer that we don't show the reset on 4. It's something that show caring about details, as the reset pollutes the UI with no benefit. I reckon we could show it and would be fine, so that's an option for me as well.

I don't think we should hide the reset on 3. That'll be confusing. Note that users can't tell apart primary & secondary filters, hence we'd run into situations such as: two primary filters and no reset VS one primary filter + one secondary filter + reset.

IMO, this logic belongs in dataviews, shouldn't be delegated to consumers.

I'll wait for @jameskoster on this.

Reply:
I think the simplest way to describe the availability of the reset button is:

If the current filter setup in the UI matches the configuration in the code the reset button is disabled. 

So if a consumer supplied 10 primary filters, the reset button only becomes active if the user specifies a value for one of those filters and/or adds a secondary filter.

I hope that answers the question?

Reply:
> I hope that answers the question?

That's good for when it becomes _active_ and it already works that way. How about when it is _visible_ (the options above)?

Reply:
As per https://github.com/WordPress/gutenberg/pull/58427#issuecomment-1916769050 I've updated 4 to also show the reset button.